### ISSUE TITLE
Storage Errors require Double Catching

### ISSUE THREAD

**Issue by justinbarak** [supabase-community/supabase-py#278]
**Describe the bug**
Instead of being able to a normal try/except to catch an error, I need to double try/except to catch the error.

**To Reproduce**
1. sto = supaClient.storage()
2. sto.from_(bucket).upload(f"{folder}/{excel_name}", excel_file_path)
3. Repeat 1
4. Get uncatchable (for a single try/except block) error

**Expected behavior**
Exception would be caught in single try/except block.

**Logged Error**
```HTTPStatusError                           Traceback (most recent call last)
File [redacted]/.venv/lib/python3.10/site-packages/storage3/_sync/file_api.py:38, in SyncBucketActionsMixin._request(self, method, url, headers, json, files)
     37 try:
---> 38     response.raise_for_status()
     39 except HTTPError:

File [redacted]/.venv/lib/python3.10/site-packages/httpx/_models.py:1508, in Response.raise_for_status(self)
   1507 message = message.format(self, error_type=error_type)
-> 1508 raise HTTPStatusError(message, request=request, response=self)

HTTPStatusError: Client error '400 Bad Request' for url 'https://[redacted].supabase.co/storage/v1/object/survey-results/results-summary/[redacted].xlsx'
For more information check: https://httpstatuses.com/400

During handling of the above exception, another exception occurred:

StorageException                          Traceback (most recent call last)
Cell In [58], line 24, in supa_save_excel(excel_file_path, caller_uuid)
     21 try:
     22 # try:
     23     # for whatever reason you need 2 try/except blocks to catch the errors it raises
---> 24     sto.from_(bucket).upload(f"{folder}/{excel_name}", excel_file_path)
     25 # except:
     26 #     pass
...
     41         {**response.json(), "statusCode": response.status_code}
     42     )
     44 return response

StorageException: {'statusCode': 400, 'error': '', 'message': 'duplicate key value violates unique constraint "bucketid_objname"'}
```

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04.3
 - Python: 3.10.7

**Additional context**
Double try/except block will catch the error, but that seems a bit absurd. It looks like httpx now raises HTTPStatusError instead of HTTPError, so it is not caught in the try block of file_api. Going to httpx it looks like that is fixed in the latest version, 0.23.0. However, Supabase has pinned a previous version <= 0.22.0. Updating that requirement might fix the issue.

**Comment by justinbarak** [comment]
It looks like I can just precede every upload with `sto.from_(bucket).remove([f"{folder}/{excel_name}"])`. That at least will not raise an error if the document doesn't exist.

**Comment by anand2312** [comment]
Hmm that is odd. Would you like to try fixing this by updating the dependency as you suggest?
Just clone the repo, modify the pyproject,toml to the new version/run `poetry update` and commit the updated files @justinbarak

**Comment by cadnce** [comment]
Looks like i cannot reproduce this anymore, I think it was probably fixed as part of ec94092189d0309fa4f1a7cfb6015ddacc1601c5

👍

**Comment by justinbarak** [comment]
> Looks like i cannot reproduce this anymore, I think it was probably fixed as part of ec94092189d0309fa4f1a7cfb6015ddacc1601c5
> 
> 
> 
> 👍 

I think you're right. I see the httpx package version updated in the poetry.toml. Closed.
